packages/foam-vscode/src/core/markdown-provider.ts

```diff
@@ -201,6 +222,53 @@ const tagsPlugin: ParserPlugin = {
   },
 };
 
+let blockStack: Array<{ label: string; level: number; start: Position }> = [];
+const blocksPlugin: ParserPlugin = {
```

Comment:
Not specific of this change, but how well does reparsing the entire file on each change scale? Do you know? I see it's an async operation, but anyway it could be pretty expensive for big files. In the ZK many-small-files scenario is not problem, but if you have some big files in the same workspace, especially now that there is section linking support, couldn't it be a problem?

Reply:
Performance is something that I have been postponing. At some point we'll need to take a serious look at it (starting from whether using a parser is better than regexes, which other tools have preferred). But I have (semi-)consciously decided to leave it for later